nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -263,6 +263,14 @@ in {
           may be served via HTTPS.
         '';
       };
+
+      extraConfig = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        description = ''
+          Extra text to append to nextcloud override config options.
+        '';
+      };
```

Comment:
I'm wondering if a `settings` option as described in NixOS/rfcs#42 would make sense here: otherwise people would accidentally override things, with such an approach, override and merging could be implemented in a better way.

Reply:
Not the most fun thing to do... but I agree `settings` would be a much better idea. See https://github.com/NixOS/nixpkgs/blob/896270d629efd47d14972e96f4fbb79fc9f45c80/nixos/modules/services/web-apps/limesurvey.nix#L23 for the least insane approach several people could come up with.

Reply:
what if you need to code a function inside config.php? it's an executable file for a reason probably, even though I'm not a Nextcloud pro

Reply:
I am using Nextcloud for a while now and newer saw this use case.